operatorcourier/cli.py

```diff
@@ -113,3 +114,24 @@ def nest(self):
 
         args, leftovers = parser.parse_known_args(sys.argv[2:])
         api.nest(args.source_dir, args.registry_dir)
+
+    # Parse the flatten command
+    def flatten(self):
```

Comment:
I think something is missing from the usage string of `flatten -h`:

```bash
$ operator-courier flatten -h
usage: operator-courier [-h] source_dir dest_dir
```

Reply:
@csomh This is strange... I am able to see the description on my side.

```
(operator-courier) 10:42 > operator-courier git:(add-flatten-command) ✗ operator-courier flatten -h
usage: operator-courier [-h] source_dir dest_dir

Given a directory with different versions of operator bundles (CRD, CSV,
package), this command extracts versioned CSVs and the latest version of each
CRD along with the package file and creates a new flat directory of yaml
files. See https://github.com/operator-framework/operator-registry#manifest-
format to find out more about how nested bundles should be structured.

positional arguments:
  source_dir  Path of the source directory that contains different versions of
              operator bundles (CRD, CSV, package)
  dest_dir    The new flat directory that contains extracted bundle files

optional arguments:
  -h, --help  show this help message and exit
```

Reply:
@jeremylinlin, it's not about the description, it's the usage string. For `operator-courier flatten -h` my expectation would be

```
operator-courier flatten [-h] source_dir dest_dir
```

but it's:

```
operator-courier [-h] source_dir dest_dir
```

Reply:
@csomh Gotcha. But it seems to be a general issue for all other subcommands. I fiddled around for some time and wasn't able to find a quick fix. Maybe we should open up an issue and make a separate PR for this? @kevinrizza

Reply:
It's a good point, but let's not check in more bad code. Let's do this one right and fix the others in a separate PR.

Reply:
@kevinrizza May I hard code the right usage for `flatten` for this time?
```python
    def flatten(self):
        parser = argparse.ArgumentParser(
            usage='operator-courier flatten [-h] source_dir dest_dir',
            description='Given a directory with different versions of '
                        'operator bundles (CRD, CSV, package), this command extracts '
                        'versioned CSVs and the latest version of each CRD along with '
                        'the package file and creates a new flat directory '
                        'of yaml files. See https://github.com/operator-framework/'
                        'operator-registry#manifest-format to find out more about '
                        'how nested bundles should be structured.')
```

Because after reading some [docs about subcommands](https://docs.python.org/3/library/argparse.html#sub-commands) and doing some experiments, it seems the current subcommands implementation is not the standard way of doing it. And it will be very hard to make the `flatten` help message work using the "right way" without breaking others.

I can also make a PR to fix the CLI subcommands before fixing this PR, but this may take a while to get through while the current PR is more urgent.